plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestTimestampCoercer.java

```diff
@@ -45,6 +48,44 @@
 
 public class TestTimestampCoercer
 {
+    @Test
+    public void testTimestampToDate()
+    {
+        // before epoch
+        assertTimestampToDateCoercion("1900-01-01T00:00:00.000", "1900-01-01");
+        assertTimestampToDateCoercion("1958-01-01T13:18:03.123", "1958-01-01");
+        // after epoch
+        assertTimestampToDateCoercion("2019-03-18T10:01:17.987", "2019-03-18");
+        // time doubled in JVM zone
+        assertTimestampToDateCoercion("2018-10-28T01:33:17.456", "2018-10-28");
+        // epoch
+        assertTimestampToDateCoercion("1970-01-01T00:00:00.000", "1970-01-01");
+        // time gap in JVM zone
+        assertTimestampToDateCoercion("1970-01-01T00:13:42.000", "1970-01-01");
+        assertTimestampToDateCoercion("2018-04-01T02:13:55.123", "2018-04-01");
+        // time gap in Vilnius
+        assertTimestampToDateCoercion("2018-03-25T03:17:17.000", "2018-03-25");
+        // time gap in Kathmandu
+        assertTimestampToDateCoercion("1986-01-01T00:13:07.000", "1986-01-01");
+        // before epoch with second fraction
+        assertTimestampToDateCoercion("1969-12-31T23:59:59.123456", "1969-12-31");
+    }
+
+    @Test
+    public void testHistoricalLongTimestampToDate()
+    {
+        LocalDateTime localDateTime = LocalDateTime.parse("1899-12-31T23:59:59.999999999");
+        SqlTimestamp timestamp = SqlTimestamp.fromSeconds(TIMESTAMP_PICOS.getPrecision(), localDateTime.toEpochSecond(UTC), localDateTime.get(NANO_OF_SECOND));
+        assertThatThrownBy(() ->
+                assertLongTimestampToDateCoercions(
+                        TIMESTAMP_PICOS,
+                        new LongTimestamp(timestamp.getEpochMicros(), timestamp.getPicosOfMicros()),
+                        DATE,
+                        -25568L))
```

Comment:
nit: Can we use this method `assertTimestampToDateCoercion(String timestampAsString, String expectedDate)` so in future if we are planning to add support for historical dates - the cleanup could be simple.